src/engines/engine-decorator.js

```diff
@@ -22,6 +22,7 @@ class EngineDecorator extends FakeEventTarget implements IEngineDecorator {
   constructor(engine: IEngine, pluginWithDecorators: Array<IEngineDecoratorProvider>) {
     super();
     this._eventManager = new EventManager();
+    this._eventManager.listen(engine, EventType.PLAYER_DESTROY, () => this._destroy());
```

Comment:
@Yuvalke why listen to the underlying engine when this is explicitly called by the player?
Add this as a method and call the underlying engine explicitly.